pandas/io/stata.py

```diff
@@ -1327,12 +1327,20 @@ def _calcsize(self, fmt):
 
     def _decode(self, s):
         s = s.partition(b"\0")[0]
-        return s.decode('utf-8')
+        try:
+            return s.decode('utf-8')
```

Comment:
Out of curiosity why is this one hard coded where other methods seem to use `self._encoding`? Just trying to keep track of the various assignments and assumptions going on here

Reply:
Agreed, some commenting regarding the logic for this would be useful.

Reply:
I refactored to remove null_terminate which isn't actually needed.